nixos/modules/services/monitoring/prometheus/exporters/ipmi.nix

```diff
@@ -0,0 +1,54 @@
+{ config, lib, pkgs, options }:
+
+with lib;
+
+let
+  logPrefix = "services.prometheus.exporter.ipmi";
+  cfg = config.services.prometheus.exporters.ipmi;
+
+  checkConfigLocation = file:
+    if file == null then
+      null
+    else if lib.hasPrefix "/tmp/" file then
+      throw
+      "${logPrefix}: configuration file must not reside within /tmp - it won't be visible to the systemd service."
+    else
+      file;
+in {
+  port = 9290;
+
+  extraOpts = {
+    configFile = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = lib.mdDoc ''
+        Path to configuration file.
+      '';
+    };
+
+    webConfigFile = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = lib.mdDoc ''
+        Path to configuration file that can enable TLS or authentication.
+      '';
+    };
+  };
+
+  serviceOpts = let
+    checkedConfigFile = checkConfigLocation cfg.configFile;
+    checkedWebConfigFile = checkConfigLocation cfg.webConfigFile;
+  in {
+    serviceConfig = {
+      ExecStart = ''
+        ${pkgs.prometheus-ipmi-exporter}/bin/ipmi_exporter \
+          --web.listen-address ${cfg.listenAddress}:${toString cfg.port} '' +
+        (optionalString (checkedWebConfigFile != null) '' --web.config.file ${escapeShellArg checkedWebConfigFile} '') +
+        (optionalString (checkedConfigFile != null) '' --config.file ${escapeShellArg checkedConfigFile} '') + '' \
+        ${concatStringsSep " \\\n  " cfg.extraFlags}
+      '';
+
+      ExecReload = "${pkgs.coreutils}/bin/kill -HUP $MAINPID";
```

Comment:
Are you sure this works? I can't find any mention of this in the repo.

Reply:
I found it in code: https://github.com/prometheus-community/ipmi_exporter/blob/bb59b211ee3fe27d3327cc31a5441fa24bb6cb1b/main.go#L121

Reply:
Alright, I see it now. Thanks for clearing this up.